reVX/least_cost_xmission/offshore_cost_creator_cli.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=all
+"""
+Xmission Cost Creator Command Line Interface
+"""
+import os
+import sys
+import click
+import logging
+from typing import Dict, Union
+
+from rex.utilities.loggers import init_logger
+
+from reVX import __version__
+from reVX.least_cost_xmission.offshore_utilities import CombineRasters
+from reVX.config.least_cost_xmission import OffshoreCreatorConfig,\
+      BarrierFiles, FrictionFiles
+
+KwargsDict = Dict[str,
+                  Union[str, int, float, bool, BarrierFiles, FrictionFiles]]
+
+logger = logging.getLogger(__name__)
+init_logger('reVX', log_level="DEBUG")
+
+# TODO - create land mask command
+
+
+@click.command
+@click.option('-c', '--config', 'config_fpath', type=click.Path(exists=True),
+              required=True, help='Configuration JSON.')
+@click.option('--create-h5/--dont-create-h5', default=True,
+              help='Create a new offshore H5 data file.')
+def from_config(config_fpath: str, create_h5: bool):
+    """
+    Create offshore barriers and frictions from a config file.
+    """
+    config = OffshoreCreatorConfig(config_fpath)
+    if create_h5 and config.ex_offshore_h5_fpath is None:
+        click.echo('ex_offshore_h5_fpath must be set unless H5 creation is '
+                   'disabled with --dont-create-h5', err=True)
+        sys.exit(1)
+
+    kwargs: KwargsDict= {}
+    if config.layer_dir is not None:
+        kwargs['layer_dir'] = config.layer_dir
+    if config.slope_barrier_cutoff is not None:
+        kwargs['slope_barrier_cutoff'] = config.slope_barrier_cutoff
+    if config.low_slope_cutoff is not None:
+        kwargs['low_slope_cutoff'] = config.low_slope_cutoff
+    if config.high_slope_friction is not None:
+        kwargs['high_slope_friction'] = config.high_slope_friction
+    if config.medium_slope_friction is not None:
+        kwargs['medium_slope_friction'] = config.medium_slope_friction
+    if config.low_slope_friction is not None:
+        kwargs['low_slope_friction'] = config.low_slope_friction
+    cr = CombineRasters(config.template_raster_fpath, **kwargs)
+
+    kwargs = {}
+    if config.slope_fpath is not None:
+        kwargs['slope_file'] = config.slope_fpath
```

Comment:
The default of `slope_file` in the `build_off_shore_barriers` function is `None`, so do we need to jump through this hoop? I think we can just say `slope_file=config.slope_fpath` and the `None` case will be handled correctly anyways, rght?

Reply:
done